src/main/java/vw/codegurutest/infra/DynamoDbChecker.java

```diff
@@ -0,0 +1,44 @@
+package vw.codegurutest.infra;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableRequest;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableResult;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+import com.amazonaws.services.dynamodbv2.model.TableStatus;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DynamoDbChecker {
+    private final AmazonDynamoDB ddbClient;
+    private long waitTimeSeconds = 5000;
+
+    public DynamoDbChecker(AmazonDynamoDB ddbClient) {
+        this.ddbClient = ddbClient;
+    }
+
+
+    public TableDescription check(String tableName) {
+        long elapsedMs;
+        long startTimeMs = System.currentTimeMillis();
+        do {
+            DescribeTableResult describe = ddbClient.describeTable(new DescribeTableRequest().withTableName(tableName));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code appears to be waiting for a resource before it runs. You could use the waiters feature to help improve efficiency. Consider using TableExists or TableNotExists. For more information, see [https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/](https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/)